docs/Running-Hadoop-MapReduce-on-Tachyon.md

```diff
@@ -3,99 +3,157 @@ layout: global
 
-If the cluster is a 2.x cluster, then these properties are not needed.
+This is the jar that you should use for the rest of this guide.
 
-# Distributing Tachyon Executables
+# Configuring Hadoop
 
-In order for the MapReduce job to be able to use files via Tachyon, we will need to distribute the
-Tachyon jar amongst all the nodes in the cluster. This will allow the TaskTracker and JobClient to
+In order for the Tachyon client jar to be available to the JobClient, you can modify
+`HADOOP_CLASSPATH` by changing `hadoop-env.sh` to:
+
+```bash
+$ export HADOOP_CLASSPATH=/<PATH_TO_TACHYON>/clients/client/target/tachyon-client-{{site.TACHYON_RELEASED_VERSION}}-jar-with-dependencies.jar
+```
+
+This allows the code that creates and submits the Job to reference Tachyon if necessary.
+
+# Distributing the Tachyon Client Jar
+
+In order for the MapReduce job to be able to read and write files in Tachyon, the Tachyon client jar
+must be distributed to all the nodes in the cluster. This allows the TaskTracker and JobClient to
 have all the requisite executables to interface with Tachyon.
 
-We are presented with three options that for distributing the jars as outlined by this
-[guide](http://blog.cloudera.com/blog/2011/01/how-to-include-third-party-libraries-in-your-map-reduce-job/)
-from Cloudera.
+This guide on
+[how to include 3rd party libraries from Cloudera](http://blog.cloudera.com/blog/2011/01/how-to-include-third-party-libraries-in-your-map-reduce-job/)
+describes several ways to distribute the jars. From that guide, the recommended way to distributed
+the Tachyon client jar is to use the distributed cache, via the `-libjars` command line option.
+Another way to distributed the client jar is to manually distributed it to all the Hadoop nodes.
+Below are instructions for the 2 main alternatives:
+
+1. **Using the -libjars command line option.**
+You can run a job by using the `-libjars` command line option when using `hadoop jar ...`, and
+specifying
+`/<PATH_TO_TACHYON>/clients/client/target/tachyon-client-{{site.TACHYON_RELEASED_VERSION}}-jar-with-dependencies.jar`
+as the argument. This will place the jar in the Hadoop DistributedCache, and should be available on
+the nodes. For example, the following command adds the Tachyon client jar to the `-libjars` option:
+
+```bash
+$ hadoop jar hadoop-examples-1.2.1.jar wordcount -libjars /<PATH_TO_TACHYON>/clients/client/target/tachyon-client-{{site.TACHYON_RELEASED_VERSION}}-jar-with-dependencies.jar <INPUT FILES> <OUTPUT DIRECTORY>`
+```
+
+1. **Distributing the jars to all nodes manually.**
+For installing Tachyon on each node, you must place the client jar
+`tachyon-client-{{site.TACHYON_RELEASED_VERSION}}-jar-with-dependencies.jar`
+(located in the `/<PATH_TO_TACHYON>/clients/client/target/` directory), in the `$HADOOP_HOME/lib`
+directory of every MapReduce node, and then restart all of the TaskTrackers. One downfall of this
```

Comment:
I used Hadoop2.4.1 in a setup, when I put tachyon client jar into `$HADOOP_HOME/share/hadoop/common`, I can call `$HADOOP_HOME/bin/hadoop jar xxxx` without specifying `-libjars`, but when I put it into `$HADOOP_HOME/lib`, `-libjars` was still needed.